src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -839,4 +838,143 @@ private CommandExecutorStat ExecuteDataFrameGroupedMapCommand(
             return stat;
         }
     }
+
+    internal class ArrowOrDataFrameCoGroupedMapCommandExecutor : ArrowOrDataFrameSqlCommandExecutor
```

Comment:
Hi Igor, is it possible to split this PR a little bit? This PR is huge, better to only contain 3.5 support and Net 8 support. We can leave other changes to future PRs.

Reply:
Sure, Dan. I'll create a separate PR for CoGrouped UDFs and binary serializer.  Vast majority of other fixes are related to each other though, so PR will still be relatively large.
At one point, I was unsure if this would ever get merged, so I ended up including all the improvements I needed to properly test whether the library meets my requirements in one place, so that if someone wants to build a version it's relatively simple to accomplish.

Reply:
@wudanzy 

https://github.com/dotnet/spark/pull/1178#issuecomment-2309533060

It will help this project if the support of polyglot notebook is included
https://github.com/dotnet/spark/pull/1178#issuecomment-2309508160

Reply:
Possible to setup CI/CD so that each PR usability can be tracked?

Reply:
> Possible to setup CI/CD so that each PR usability can be tracked? @SparkSnail https://github.com/SparkSnail/spark/actions

Yes, current test pipeline for the repo is broken, we are working to recover pipeline for PRs.

Reply:
Removed unnecessary refactoring and new features from the PR, preserved only .NET, Spark 3.5 and a few fixes